web/src/migrating/import/import.ts

```diff
@@ -1,14 +1,26 @@
+import { z } from 'zod'
 import { getAppWs } from '../../hcWebsockets'
 import { cellIdFromString } from '../../utils'
 import { RootState } from '../../redux/reducer'
-import { AllProjectsDataExport } from '../export'
 import { createWhoami } from '../../redux/persistent/profiles/who-am-i/actions'
 import { installProjectApp } from '../../projects/installProjectApp'
 import { joinProjectCellId } from '../../redux/persistent/cells/actions'
 import { createProfilesZomeApi, createProjectsZomeApi } from './zomeApiCreators'
 import { installProjectAppAndImport } from './installProjectAppAndImport'
 import ProfilesZomeApi from '../../api/profilesApi'
 import ProjectsZomeApi from '../../api/projectsApi'
+import { AllProjectsDataExport } from '../model/allProjectsDataExport'
+
+const stringToJSONSchema = z.string().transform(
+  (str, ctx): AllProjectsDataExport => {
+    try {
+      return JSON.parse(str)
+    } catch (e) {
+      ctx.addIssue({ code: 'custom', message: 'Invalid JSON' })
+      return z.NEVER
+    }
+  }
+)
```

Comment:
Yeah good point. I suppose it's technically just a way of zod wrapping the JSON.parse(), and then the runtime tries to coerce the valid json into typeof AllProjectsDataExport.

The only thing is, right now it isn't really possible to get zod to validate the json for us, because without strict mode enabled, the returned data from AllProjectsDatExportSchema.parse() is all optional, so typescript complains because the schema will allow the data to be incomplete.

the only way to fix this would be to find a way to enable strict mode, or as you suggested above, to put all the schemas in their own npm package inside the project so that we can enable strict mode without interfering with the rest of the code.

Reply:
I think we may want and need to take that approach of creating its own package inside the project

Reply:
so i just tried this. i created another npm package within the `web` folder, and copied all the schema definitions over to the new package and created an inferred type from the schema.

unfortunately, when i import this new schema into the parent web project, it goes back to having all optional fields :( i can only assume that zod's inferred types *literally* only work when strict mode is true. we might need to come up with a different solution.

Reply:
maybe we just need to bite the bullet and go through the whole project and fix all the type errors @Connoropolous